### PR TITLE
Feature/rebuild ir unit test

### PR DESCRIPTION
**Pull request by Hejunchao6**

### CODE REVIEW

**Review by Hejunchao6 — COMMENTED** [review]
review

File: src/Nncase.Tests/Core/UnitTestQuantUtility.cs [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+using Xunit;
+
+namespace Nncase.Tests.CoreTest;
+
+public sealed class UnitTestQuantUtility
+{
+    [Fact]
+    public void TestGetQuantParam()
+    {
+        var quantParam1 = QuantUtility.GetQuantParam(new ValueRange<float>(-1f, 1f), 1, QuantMode.SignedSymmetricMode);
```
Comment (xhuohai) [inline src/Nncase.Tests/Core/UnitTestQuantUtility.cs:19]: change bits to 8 or 16, and change scale accordingly

File: src/Nncase.Tests/Core/UnitTestQuantUtility.cs [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        Assert.Equal(new QuantParam(0, float.PositiveInfinity), quantParam1);
+
+        var quantParam2 = QuantUtility.GetQuantParam(new ValueRange<float>(-1f, 1f), 1, QuantMode.SignedAsymmetricMode);
```
Comment (xhuohai) [inline src/Nncase.Tests/Core/UnitTestQuantUtility.cs:22]: same as above

File: src/Nncase.Tests/Core/UnitTestQuantUtility.cs [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+        Assert.Equal(new QuantParam(-1, 2f), quantParam2);
+    }
+
+    [Fact]
+    public void TestFixupRange()
+    {
+        var range1 = QuantUtility.FixupRange(new ValueRange<float>(0f, 0f));
+        Assert.Equal(new ValueRange<float>(0f, 0.1f), range1);
+
+        var range2 = QuantUtility.FixupRange(new ValueRange<float>(0f, 0.005f));
```
Comment (xhuohai) [inline src/Nncase.Tests/Core/UnitTestQuantUtility.cs:32]: should add negative ranges

File: src/Nncase.Tests/Evaluator/UnitTestEvaluator.cs [hunk 4]
```diff
@@ -91,4 +92,40 @@ public void TestOnnxResizeImage()
         image.InferenceType();
         Assert.Equal(new[] { 1, 3, 112, 112 }, image.Evaluate().AsTensor().Dimensions.ToArray());
     }
+
+    [Fact]
+    public void TestLoadStore()
+    {
+        var loop_i = new Var(TensorType.Scalar(DataTypes.Int32));
+        var load = T.Load(T.Handle("hd", DataTypes.Float32), loop_i);
+        CompilerServices.InferenceType(load);
```
Comment (xhuohai) [inline src/Nncase.Tests/Evaluator/UnitTestEvaluator.cs:101]: should compare results

File: src/Nncase.Tests/Evaluator/UnitTestEvaluator.cs [hunk 5]
```diff
@@ -91,4 +92,40 @@ public void TestOnnxResizeImage()
...
+    {
+        var nop = T.Nop();
+        CompilerServices.InferenceType(nop);
+    }
+
+    [Fact]
+    public void TestRamp()
+    {
+        var ramp = T.Ramp(1, 2, 0);
+        CompilerServices.InferenceType(ramp);
```
Comment (xhuohai) [inline src/Nncase.Tests/Evaluator/UnitTestEvaluator.cs:118]: save as above

File: src/Nncase.Tests/Quant/UnitTestPytestCalibrationDatasetProvider.cs [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+public class UnitTestPytestCalibrationDatasetProvider
+{
+    [Fact]
+    public void TestPytestCalibrationDatasetProvider()
+    {
+        var input = new Var();
+        var vars = new[] { input };
+        var dataset = "./";
+        var provider = new PytestCalibrationDatasetProvider(vars, dataset);
+        Assert.Equal(0, provider.Count);
```
Comment (xhuohai) [inline src/Nncase.Tests/Quant/UnitTestPytestCalibrationDatasetProvider.cs:29]: should change

File: src/Nncase.Tests/Quant/UnitTestRandomCalibrationDatasetProvider.cs [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+
+public class UnitTestRandomCalibrationDatasetProvider
+{
+    [Fact]
+    public void TestRandomCalibrationDatasetProvider()
+    {
+        var input = new Var();
+        var vars = new[] { input };
+        var provider = new RandomCalibrationDatasetProvider(vars, 1);
+        Assert.Equal(1, provider.Count);
```
Comment (xhuohai) [inline src/Nncase.Tests/Quant/UnitTestRandomCalibrationDatasetProvider.cs:28]: should test data generation

File: src/Nncase.Tests/Quant/UnitTestUtility.cs [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+
+public class UnitTestUtility
+{
+    [Fact]
+    public void TestGetCosineSimilarity()
+    {
+        var v1 = new float[] { 1f, 2f, 3f, 4f };
+        var v2 = new float[] { 1f };
+        var actual = Utility.GetCosineSimilarity(v1, v2);
+        Assert.Equal(1f, actual);
```
Comment (xhuohai) [inline src/Nncase.Tests/Quant/UnitTestUtility.cs:28]: cos sim is 1 for two different vectors?

File: src/Nncase.Tests/Quant/UnitTestUtility.cs [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+        Assert.Equal(1f, Utility.GetCosineSimilarity(new[] { 0f }, new[] { 0f }));
+    }
+
+    [Fact]
+    public void TestGetFixedMul()
+    {
+        var actual1 = Utility.GetFixedMul(0f, 0, 0, true);
+        var actual2 = Utility.GetFixedMul(2f, 0, 0, true);
```
Comment (xhuohai) [inline src/Nncase.Tests/Quant/UnitTestUtility.cs:36]: shift and bits shoud change to significant value
sign also need to test `false`

**Review by xhuohai — APPROVED** [review]

**Pull request merged**